controllers/module_reconciler.go

```diff
@@ -440,9 +428,55 @@ func (mrh *moduleReconcilerHelper) setKMMOMetrics(ctx context.Context) {
 	err := mrh.client.List(ctx, &mods)
 	if err != nil {
 		logger.V(1).Info("failed to list KMMomodules for metrics", "error", err)
-	}
+		return
+	}
+
+	numModules := len(mods.Items)
+	numModulesWithBuild := 0
+	numModulesWithSign := 0
+	numModulesWithDevicePlugin := 0
+	for _, mod := range mods.Items {
+		signFound := false
+		buildFound := false
+		if mod.Spec.DevicePlugin != nil {
+			numModulesWithDevicePlugin += 1
+		}
+		if mod.Spec.ModuleLoader.Container.Build != nil {
+			buildFound = true
+		}
+		if mod.Spec.ModuleLoader.Container.Sign != nil {
+			signFound = true
+		}
+		if !buildFound || !signFound {
+			for _, mapping := range mod.Spec.ModuleLoader.Container.KernelMappings {
+				if mapping.Sign != nil {
+					signFound = true
+				}
+				if mapping.Build != nil {
+					buildFound = true
+				}
+			}
+		}
+		if buildFound {
+			numModulesWithBuild += 1
+		}
+		if signFound {
+			numModulesWithSign += 1
+		}
 
-	mrh.metricsAPI.SetExistingKMMOModules(len(mods.Items))
+		if mod.Spec.ModuleLoader.Container.Modprobe.Args != nil {
+			modprobeArgs := strings.Join(mod.Spec.ModuleLoader.Container.Modprobe.Args.Load, ",")
+			mrh.metricsAPI.SetKMMOModprobeArgs(mod.Name, mod.Namespace, modprobeArgs)
+		}
+		if mod.Spec.ModuleLoader.Container.Modprobe.RawArgs != nil {
+			modprobeRawArgs := strings.Join(mod.Spec.ModuleLoader.Container.Modprobe.RawArgs.Load, ",")
+			mrh.metricsAPI.SetKMMOModprobeRawArgs(mod.Name, mod.Namespace, modprobeRawArgs)
+		}
+	}
+	mrh.metricsAPI.SetKMMOModulesNum(numModules)
+	mrh.metricsAPI.SetKMMOInClusterBuildNum(numModulesWithBuild)
+	mrh.metricsAPI.SetKMMOInClusterSignNum(numModulesWithSign)
+	mrh.metricsAPI.SetKMMODevicePluginNum(numModulesWithDevicePlugin)
```

Comment:
I would call `mrh.metricsAPI.SetKMMOModprobeArgs` here as well instead for code clarity.